common/io/TimeoutManager.cpp

```diff
@@ -97,8 +97,7 @@ TimeInterval TimeoutManager::ExecuteTimeouts(TimeStamp *now) {
   if (m_events.empty())
     return TimeInterval();
 
-  for (e = m_events.top(); !m_events.empty() && (e->NextTime() <= *now);
-       e = m_events.top()) {
+  while (!m_events.empty() && ((e = m_events.top())->NextTime() <= *now)) {
```

Comment:
Can we add a comment for our future selves please, something like this:

```suggestion
  // Use a while loop rather than a for loop, so the iteration expression
  // assigning m_events.top() isn't executed after the loop on a potentially
  // empty m_events queue.
  while (!m_events.empty() && ((e = m_events.top())->NextTime() <= *now)) {
```